src/base/bittorrent/torrentimpl.cpp

```diff
@@ -194,9 +139,30 @@ namespace
                 firstTrackerMessage = trackerMessage;
             if (firstErrorMessage.isEmpty())
                 firstErrorMessage = errorMessage;
-        }
+
+            return trackerEndpoint;
+        };
+
+        for (const lt::announce_endpoint &endpoint : nativeEntry.endpoints)
+        {
+#ifdef QBT_USES_LIBTORRENT2
+            for (const auto protocolVersion : {lt::protocol_version::V1, lt::protocol_version::V2})
+            {
+                if (!hashes.has(protocolVersion))
+                    continue;
+                trackerEntry.stats[endpoint.local_endpoint][(protocolVersion == lt::protocol_version::V1) ? 1 : 2] =
```

Comment:
In case it will be approved...
I see no point to allow `makeEndpoint()` behave inconsistently, i.e. modify some fields of TrackerEntry directly and other one by assigning return value from outside of it. Why not assign `trackerEntry.stats` inside `makeEndpoint()`?

Reply:
makeEndpoint() is common version agnostic code. Only branching (and difference) between versions is here.

Reply:
> `makeEndpoint()` is common version agnostic code. Only branching (and difference) between versions is here.

Sorry, I don't see any reasons justifying writing confusing code. IMO, if a statement like `obj.m1 = f(a, b);` also changes `obj.m2`, it is "bad" code.